apps/build-notification/src/githubPrComment.ts

```diff
@@ -0,0 +1,132 @@
+import type { Octokit } from "@octokit/rest";
+
+import { Build, type PullRequest } from "@argos-ci/database/models";
+import { getRedisLock } from "@argos-ci/web";
+
+export const getAggregateStatuses = async (builds: Build[]) => {
+  const buildIds = builds.map((build) => build.id);
+  const statuses = await Build.getStatuses(builds);
+  const conclusions = await Build.getConclusions(buildIds, statuses);
+  const reviewStatuses = await Build.getReviewStatuses(buildIds, conclusions);
+  return builds.map((_build, index) => {
+    if (reviewStatuses[index]) return reviewStatuses[index];
+    if (conclusions[index]) return conclusions[index];
+    return statuses[index];
+  });
+};
+
+export const formatStatus = (status: string | null | undefined) => {
+  switch (status) {
+    case "stable":
+      return "✅ no diff detected";
+
+    case "accepted":
+      return "👍 approuved";
+
+    case "error":
+    case "expired":
+    case "rejected":
+    case "aborted":
+      return `❌ ${status}`;
+
+    case "pending":
+    case "progress":
+      return "⏳ pending";
```

Comment:
"pending" and "progress" are not the same thing:

- Pending: waiting for screenshots
- Progress: diffing screenshots